x/posts/types/models.go

```diff
@@ -599,7 +616,52 @@ func (a Poll_ProvidedAnswer) Validate() error {
 		return fmt.Errorf("invalid text: %s", a.Text)
 	}
 
-	return Attachments(a.Attachments).Validate()
+	// Unpack the attachments
+	attachments := make([]AttachmentContent, len(a.Attachments))
+	for i, attachmentAny := range a.Attachments {
+		attachments[i] = attachmentAny.GetCachedValue().(AttachmentContent)
+	}
+
+	// Validate the attachments
+	for _, attachment := range attachments {
+		if _, isPoll := attachment.(*Poll); isPoll {
+			return fmt.Errorf("cannot have a poll as an attachment of a poll's provided answer")
+		}
+
+		if containsDuplicated(attachments, attachment) {
+			return fmt.Errorf("duplicated attachment")
+		}
+
+		err := attachment.Validate()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// containsDuplicated tells whether the given attachments array contains a duplicate content
+func containsDuplicated(attachments []AttachmentContent, content AttachmentContent) bool {
```

Comment:
I would name it `containsDuplicatedContents`

Reply:
I've renamed this `containsDuplicatedAttachments` since it checks if there are duplicated attachments within the provided answer.